oscrypto/_openssl/_libcrypto_cffi.py

```diff
@@ -27,10 +27,12 @@
 libcrypto_path = _backend_config().get('libcrypto_path')
 if libcrypto_path is None:
     libcrypto_path = find_library('crypto')
+    # If we are on macOS 10.16+, find_library doesn't work, so we set a static path
+    if sys.platform == 'darwin' and tuple(map(int,  platform.mac_ver()[0].split('.'))) >= (10, 16):
```

Comment:
So here we should check if `libcrypto_path` is empty. The code below had a bug because always assumed a system-level location, but some users have run it in conda, etc.

Reply:
You mean like this? Still keeping the logic for MacOS 10.15+? 

```diff
libcrypto_path = _backend_config().get('libcrypto_path')
if libcrypto_path is None:
    libcrypto_path = find_library('crypto')
+    if sys.platform == 'darwin' and not libcrypto_path:
+        libcrypto_path == '/usr/lib/libcrypto.42.dylib'
-    # If we are on macOS 10.16+, find_library doesn't work, so we set a static path
-    if sys.platform == 'darwin' and tuple(map(int,  platform.mac_ver()[0].split('.'))) >= (10, 16):
-        libcrypto_path == '/usr/lib/libcrypto.42.dylib'
-    # if we are on macOS 10.15+, we want to strongly version libcrypto since unversioned libcrypto has a non-stable ABI
-    if sys.platform == 'darwin' and tuple(map(int,  platform.mac_ver()[0].split('.'))) >= (10, 15) and \
-            libcrypto_path == '/usr/lib/libcrypto.dylib':
-        libcrypto_path = libcrypto_path.replace('libcrypto.dylib', 'libcrypto.42.dylib')
if not libcrypto_path:
    raise LibraryNotFoundError('The library libcrypto could not be found')
```

Reply:
No, just:

```suggestion
    if not libcrypto_path and sys.platform == 'darwin' and tuple(map(int,  platform.mac_ver()[0].split('.'))) >= (10, 16):
```

That way we only hard code it for it being missing on 10.16. We know that version will be there on 10.16, but it doesn't exist on older versions of macOS.